azext_iot/digitaltwins/providers/resource.py

```diff
@@ -71,6 +74,25 @@ def create(
                 long_running_operation_timeout=timeout,
             )
 
+            def check_state(lro):
+                from time import sleep
+                instance = lro.resource().as_dict()
+                state = instance.get('provisioning_state', None)
+                retries = 0
+                while (state.lower() not in ProvisioningStateType.FINISHED.value) and retries < MAX_ADT_CREATE_RETRIES:
+                    retries += 1
+                    sleep(ADT_CREATE_RETRY_AFTER)
+                    lro.update_status()
+                    instance = lro.resource().as_dict()
+                    state = instance.get('provisioning_state', None)
+                if state.lower() not in ProvisioningStateType.FINISHED.value:
```

Comment:
Is it possible that somehow `provisioning_state` doesn't exist on this object? Just a concern that if we retry `MAX_ADT_CREATE_RETRIES` times, and something happens to the `provisioning_state`:

`state = instance.get('provisioning_state', None)` -> assigns `None` to `state`

`if state.lower()` -> `AttributeError: 'NoneType' object has no attribute 'lower'`

Maybe we just need `if state and state.lower()`?

Reply:
Technically, there should always be a provisioning_state. However, will change it to be state and state.lower in case there is some weirdness